### PR TITLE
Fix some documentation in ./src/diffusers/models/embeddings.py for demo

### PR DESCRIPTION
**Pull request by DTG2005**
# What does this PR do?

Fixes #9567 

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

This PR tries to attempt a solution at one of the submodules listed in #9567 so I think @a-r-r-o-w is the best to review it. Alongside the same, @charchit7 @yijun-lee and @SubhasmitaSw were also working on the same, so just a ping for the update on the same.

### CODE REVIEW

File: src/diffusers/models/embeddings.py [hunk 1]
```diff
@@ -129,8 +138,24 @@ def get_2d_sincos_pos_embed(
...
+        Whether or not to add a classification token.
+    extra_tokens: int
+        The number of extra tokens to add.
+    interpolation_scale: float
+        The scale of the interpolation.
+
+    Returns:
+        pos_embed: np.ndarray
+            Shape is [grid_size*grid_size, embed_dim] or
+            [1+grid_size*grid_size, embed_dim] (w/ or w/o cls_token)
```
Comment (a-r-r-o-w) [inline src/diffusers/models/embeddings.py:158]:
```suggestion
    Creates 2D sinusoidal positional embeddings.

    Args:
        embed_dim (`int`):
            The embedding dimension.
        grid_size (`int`):
            The size of the grid height and width.
        cls_token (`bool`, defaults to `False`):
            Whether or not to add a classification token.
        extra_tokens (`int`, defaults to `0`):
            The number of extra tokens to add.
        interpolation_scale (`float`, defaults to `1.0`):
            The scale of the interpolation.

    Returns:
        pos_embed (`np.ndarray`):
            Shape is either `[grid_size * grid_size, embed_dim]` if not using cls_token, or
            `[1 + grid_size*grid_size, embed_dim]` if using cls_token
```

File: src/diffusers/models/embeddings.py [hunk 2]
```diff
@@ -148,6 +173,15 @@ def get_2d_sincos_pos_embed(
...
 def get_2d_sincos_pos_embed_from_grid(embed_dim, grid):
+    """
+    This function generates 2D positional embeddings from a grid.
+
+    Args:
+        embed_dim (`int`): output dimension for each position
+        grid (`np.ndarray`): grid of positions
+    Output:
+        `np.ndarray`: tensor in shape (grid_size*grid_size, embed_dim)
+    """
```
Comment (a-r-r-o-w) [inline src/diffusers/models/embeddings.py:184]:
```suggestion
    r"""
    This function generates 2D sinusoidal positional embeddings from a grid.

    Args:
        embed_dim (`int`): The embedding dimension.
        grid (`np.ndarray`): Grid of positions with shape `(H * W,)`.
    
    Returns:
        `np.ndarray`: The 2D sinusoidal positional embeddings with shape `(H * W, embed_dim)`
    """
```

File: src/diffusers/models/embeddings.py [hunk 3]
```diff
@@ -161,7 +195,13 @@ def get_2d_sincos_pos_embed_from_grid(embed_dim, grid):
...
 def get_1d_sincos_pos_embed_from_grid(embed_dim, pos):
     """
-    embed_dim: output dimension for each position pos: a list of positions to be encoded: size (M,) out: (M, D)
+    This function generates 1D positional embeddings from sin and cos values.
+
+    Args:
+        embed_dim(`int`): output dimension for each position
+        pos(`numpy.ndarray(dtype=float)`): tensor in shape (M, 1)
+    Output:
+        `numpy.ndarray(dtype=float)`: tensor in shape (M, D)
```
Comment (a-r-r-o-w) [inline src/diffusers/models/embeddings.py:204]:
```suggestion
    This function generates 1D positional embeddings from a grid.

    Args:
        embed_dim (`int`): The embedding dimension `D`
        pos (`numpy.ndarray`): 1D tensor of positions with shape `(M,)`
    
    Returns:
        `numpy.ndarray`: Sinusoidal positional embeddings of shape `(M, D)`.
```

File: src/diffusers/models/embeddings.py [hunk 4]
```diff
@@ -86,12 +86,21 @@ def get_3d_sincos_pos_embed(
...
         spatial_size (`int` or `Tuple[int, int]`):
+            The spatial size of the embed.
         temporal_size (`int`):
+            The temporal size of the embed.
         spatial_interpolation_scale (`float`, defaults to 1.0):
+            The spatial interpolation scale of the embed.
         temporal_interpolation_scale (`float`, defaults to 1.0):
+            The temporal interpolation scale of the embed.
+    Returns:
+        np.ndarray: The 3D position embedding.
```
Comment (a-r-r-o-w) [inline src/diffusers/models/embeddings.py:103]:
```suggestion
    Creates 3D sinusoidal positional embeddings.

    Args:
        embed_dim (`int`):
            The embedding dimension of inputs. It must be divisible by 16.
        spatial_size (`int` or `Tuple[int, int]`):
            The spatial dimension of positional embeddings. If an integer is provided, the same size is applied
            to both spatial dimensions (height and width).
        temporal_size (`int`):
            The temporal dimension of postional embeddings (number of frames).
        spatial_interpolation_scale (`float`, defaults to 1.0):
            Scale factor for spatial grid interpolation.
        temporal_interpolation_scale (`float`, defaults to 1.0):
            Scale factor for temporal grid interpolation.
    
    Returns:
        `np.ndarray`:
            The 3D sinusoidal positional embeddings of shape `[temporal_size, spatial_size[0] * spatial_size[1], embed_dim]`.
```

**Review by DTG2005 — COMMENTED** [review]
Reviewed! I'll remember the convention and follow the same when submitting future PRs!

**Review by a-r-r-o-w — APPROVED** [review]
One last thing, could you run `make style` to fix the quality tests?

**Comment by a-r-r-o-w** [comment]
Gentle ping to run `make style` and push the changes @DTG2005

**Comment by DTG2005** [comment]
Sorry for the late response, there was some college work piled up.

So while running the same after eliminating certain errors, and then some, this output is printed

```
ruff check examples scripts src tests utils benchmarks setup.py --fix
All checks passed!
ruff format examples scripts src tests utils benchmarks setup.py
15 files reformatted, 1081 files left unchanged
doc-builder style src/diffusers docs/source --max_len 119
make: doc-builder: No such file or directory
make: *** [Makefile:59: style] Error 127
```

Is this correct or should I fix some more things before I push?

**Comment by yiyixuxu** [comment]
thank you!

**Pull request merged**